test/Sema/diag_optional_to_any.swift

```diff
@@ -0,0 +1,60 @@
+// RUN: %target-parse-verify-swift
+
+func takeAny(_ left: Any, _ right: Any) -> Int? {
+  return left as? Int
+}
+
+func throwing() throws -> Int? {}
+
+func warnOptionalToAnyCoercion(value x: Int?) -> Any {
+  let a: Any = x // expected-warning {{expression implicitly coerced from 'Int?' to Any}}
+  // expected-note@-1 {{provide a default value to avoid this warning}}
+  // expected-note@-2 {{force-unwrap the value to avoid this warning}}
+  // expected-note@-3 {{explicitly cast to Any with 'as Any' to silence this warning}}
+
+  let b: Any = x as Any
+
+  let c: Any = takeAny(a, b) // expected-warning {{expression implicitly coerced from 'Int?' to Any}}
+  // expected-note@-1 {{provide a default value to avoid this warning}}
+  // expected-note@-2 {{force-unwrap the value to avoid this warning}}
+  // expected-note@-3 {{explicitly cast to Any with 'as Any' to silence this warning}}
+
+  let _: Any = takeAny(c, c) as Any
+
+  let _: Any = (x)  // expected-warning {{expression implicitly coerced from 'Int?' to Any}}
+  // expected-note@-1 {{provide a default value to avoid this warning}}
+  // expected-note@-2 {{force-unwrap the value to avoid this warning}}
+  // expected-note@-3 {{explicitly cast to Any with 'as Any' to silence this warning}}
+
+  let f: Any = (x as Any)
+  let g: Any = (x) as (Any)
+
+  _ = takeAny(f as? Int, g) // expected-warning {{expression implicitly coerced from 'Int?' to Any}}
+  // expected-note@-1 {{provide a default value to avoid this warning}}
```

Comment:
I'd like at least one of these note-cascades to test the Fix-Its themselves using the `-verify` syntax, e.g.,

    // expected-note@-1 {{provide a default value to avoid this warning}}{{23-23= ?? <#default value#>}}

to make sure the locations are reasonable and don't break in the future.